R/class-chart-sheet.R

```diff
@@ -65,24 +51,99 @@ wbChartSheet <- R6::R6Class(
+    #' @param zoomScale zoomScale
+    #' @param zoomScaleNormal zoomScaleNormal
+    #' @param zoomScalePageLayoutView zoomScalePageLayoutView
+    #' @param zoomScaleSheetLayoutView zoomScaleSheetLayoutView
+    #' @return The `wbWorksheetObject`, invisibly
+    set_sheetview = function(
+      colorId                  = NULL,
+      defaultGridColor         = NULL,
+      rightToLeft              = NULL,
+      showFormulas             = NULL,
+      showGridLines            = NULL,
+      showOutlineSymbols       = NULL,
+      showRowColHeaders        = NULL,
+      showRuler                = NULL,
+      showWhiteSpace           = NULL,
+      showZeros                = NULL,
+      tabSelected              = NULL,
+      topLeftCell              = NULL,
+      view                     = NULL,
+      windowProtection         = NULL,
+      workbookViewId           = NULL,
+      zoomScale                = NULL,
+      zoomScaleNormal          = NULL,
+      zoomScalePageLayoutView  = NULL,
+      zoomScaleSheetLayoutView = NULL
+    ) {
+
+      # all zoom scales must be in the range of 10 - 400
+
+      # get existing sheetView
+      sheetView <- xml_node(self$sheetViews, "sheetViews", "sheetView")
+
+      if (length(sheetView) == 0)
+        sheetView <- xml_node_create("sheetView")
+
+      sheetView <- xml_attr_mod(
+        sheetView,
+        xml_attributes = c(
+          colorId                  = as_xml_attr(colorId),
+          defaultGridColor         = as_xml_attr(defaultGridColor),
+          rightToLeft              = as_xml_attr(rightToLeft),
+          showFormulas             = as_xml_attr(showFormulas),
+          showGridLines            = as_xml_attr(showGridLines),
+          showOutlineSymbols       = as_xml_attr(showOutlineSymbols),
+          showRowColHeaders        = as_xml_attr(showRowColHeaders),
+          showRuler                = as_xml_attr(showRuler),
+          showWhiteSpace           = as_xml_attr(showWhiteSpace),
+          showZeros                = as_xml_attr(showZeros),
+          tabSelected              = as_xml_attr(tabSelected),
+          topLeftCell              = as_xml_attr(topLeftCell),
+          view                     = as_xml_attr(view),
+          windowProtection         = as_xml_attr(windowProtection),
+          workbookViewId           = as_xml_attr(workbookViewId),
+          zoomScale                = as_xml_attr(zoomScale),
+          zoomScaleNormal          = as_xml_attr(zoomScaleNormal),
+          zoomScalePageLayoutView  = as_xml_attr(zoomScalePageLayoutView),
+          zoomScaleSheetLayoutView = as_xml_attr(zoomScaleSheetLayoutView)
+        )
+      )
+
+      self$sheetViews <- xml_node_create(
+        "sheetViews",
+        xml_children = sheetView
+      )
+
+      invisible(self)
     }
   )
-)
-
-#' New chart sheet
-#'
-#' Create a new chart sheet
-#'
-#' @returns a `wbChartSheet` object
-#' @export
-wb_chart_sheet <- function() {
-  wbChartSheet$new()
-}
+)
```

Comment:
needs a new line and this wb function was never used, still should be mentioned in NEWS